From e73cb71d212939c2d3abccf5661c9f56a66720cf Mon Sep 17 00:00:00 2001
From: Sharath Srikanth Chellappa <sharathsr@microsoft.com>
Date: Fri, 8 Aug 2025 14:05:00 -0700
Subject: [PATCH] Enforcing webhookserver to use TLS 1.2

This patch introduces TLS configuration enforcement for the webhook server in main.go. 
Specifically, it ensures that the server uses TLS 1.2 or higher.

PR has been merged upstream - https://github.com/nunnatsa/cluster-api-provider-kubevirt/commit/8c739ccf4f39712241c9aabbf7ed3504612ecdc6

---
 main.go | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/main.go b/main.go
index 4a2fe89..8ef2a92 100644
--- a/main.go
+++ b/main.go
@@ -18,6 +18,7 @@ package main
 
 import (
 	"context"
+	"crypto/tls"
 	"flag"
 	"math/rand"
 	"os"
@@ -142,7 +143,15 @@ func main() {
 			DefaultNamespaces: defaultNamespaces,
 		},
 		HealthProbeBindAddress: healthAddr,
-		WebhookServer:          webhook.NewServer(webhook.Options{Port: webhookPort, CertDir: webhookCertDir}),
+		WebhookServer: webhook.NewServer(webhook.Options{
+			Port:    webhookPort,
+			CertDir: webhookCertDir,
+			TLSOpts: []func(*tls.Config){
+				func(t *tls.Config) {
+					t.MinVersion = tls.VersionTLS12
+				},
+			},
+		}),
 	})
 	if err != nil {
 		setupLog.Error(err, "unable to start manager")
-- 
2.49.0

